src/Api/FingerprintApi.php

```diff
@@ -127,8 +122,10 @@ public function getEvent($request_id)
 
             try {
                 $serialized = ObjectSerializer::deserialize($responseBody, $returnType, []);
-            } catch (\Exception $e) {
-                throw new SerializationException($response, $e);
+            } catch (SerializationException $e) {
+                $e->setResponse($response);
+
+                throw $e;
```

Comment:
Let's refactor `deserialize` method so that:
- it will take `ResponseInterface` parameter instead of response body
- then, we can throw `SerializeException` in it that will already contain the response
- thanks to that we can remove this nested try catch block here, on line 179-182 and in other methods